[algoliasearch] change module format

Haroenv opened the pull request:
TBH I'm not really sure what the difference is, but this doesn't break any tests on my repositories. It was needed to work with `/// reference` imports (cc @samuelbodin)

I also had to remove the Omit type for some reason to rely on the builtin one, I guess that's not too big of a deal? This does mean that we updated the TS version for algoliasearch, algoliasearch-helper, react-instantsearch, react-instantsearch-core, react-instantsearch-native, react-instantsearch-dom

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

samuelbodin commented:
> 
> 
> TBH I'm not really sure what the difference is, but this doesn't break any tests on my repositories. It was needed to work with `/// reference` imports (cc @samuelbodin)
> 
> I also had to remove the Omit type for some reason to rely on the builtin one, I guess that's not too big of a deal? This does mean that we updated the TS version for algoliasearch, algoliasearch-helper, react-instantsearch, react-instantsearch-core, react-instantsearch-native, react-instantsearch-dom
> 
> Please fill in this template.
> 
>     * [x]  Use a meaningful title for the pull request. Include the name of the package modified.
> 
>     * [x]  Test the change in your own code. (Compile and run.)
> 
>     * [x]  Add or edit tests to reflect the change. (Run with `npm test`.)
> 
>     * [x]  Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
> 
>     * [x]  Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
> 
>     * [x]  Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
> 
> 
> If changing an existing definition:
> 
>     * [x]  If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
> 
>     * [x]  If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

I think you may have cc'd wrong person :P

Haroenv commented:
Sorry @samuelbodin, I meant @bodinsamuel :)

Haroenv commented:
I'd let a TS maintainer decide on this, but I think it's on users to stay updated

keichinger commented:
That's a fair assumption, I guess. 👍

jessetrinity commented:
@Haroenv @keichinger WRT the version, without diving in too deep...

> If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Since you marked this as done, did you mean to update the types version for algiosearch which is [3.33 on npmjs](https://www.npmjs.com/package/algoliasearch), or did you really mean the TypeScript version?

Haroenv commented:
@jessetrinity, I don’t think there are methods missing, but those can be done in another PR. I checked the box because I think the type is complete, but lmk or make a pr if something is missing

jessetrinity commented:
My comment was about the fact that you updated the typescript version. This represents a minimum version of typescript required. Usually you don't want to do this unless you absolutely have to use features from the latest version of typescript.

Regarding the types version - I was speculating that you had probably intended to update that version *instead of* the typescript version.

Haroenv commented:
I updated because the original type that was written inline "Omit" was always failing, unless I removed it & bumped the TS version. I'm totally happy to change that back to something else without bumping TS

sandersn commented on types/algoliasearch/index.d.ts:
```diff
@@ -2099,4 +2097,6 @@ interface AlgoliaStatic extends AlgoliasearchInstance {
 declare const algoliasearch: AlgoliaStatic;
 
-export = algoliasearch;
+declare module 'algoliasearch' {
```
this is the old way of making a browser-accessible module. Can you please try this instead:

```ts
export = algoliasearch;
export as namespace algoliasearch;
```

This should allow you to keep your local `Omit` type, and therefore not have to update all the dependents of algoliasearch.

Haroenv replied:
Thanks @sandersn, this works perfectly ❤️

Haroenv commented on types/react-instantsearch/index.d.ts:
```diff
@@ -5,6 +5,6 @@
 //                 Samuel Vaillant <https://github.com/samouss>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.9
+// TypeScript Version: 3.5
```
```suggestion
// TypeScript Version: 2.9
```

Haroenv commented:
Thanks Nathan!
